hydradx/other tests/test_price_manipulations.py

```diff
@@ -0,0 +1,479 @@
+        tokens=tokens,
+        lrna_fee=0,
+        asset_fee=0,
+        preferred_stablecoin='USD',
+        max_withdrawal_per_block=0.05,
+        remove_liquidity_volatility_threshold=0.01,
+        max_lp_per_block=0.05,
+        trade_limit_per_block=0.05
+    )
+
+    agent_holdings = {
+        tkn: 1000000 / oamm.usd_price(initial_state, tkn) for tkn in initial_state.asset_list
+    }
+
+    initial_agent = Agent(
+        holdings=agent_holdings
+    )
+
+    lp_percent = 0.50
+    lp_token = 'myn'
+    trade_token = 'USD'
+    lp_quantity = int(initial_state.liquidity[lp_token] * lp_percent)
+
+    initial_agent.holdings[('omnipool', lp_token)] = lp_quantity
+    initial_agent.share_prices[('omnipool', lp_token)] = 0.037835736649170025
+
+    market_prices = {tkn: oamm.usd_price(initial_state, tkn) for tkn in initial_state.asset_list}
+
+    # trade to manipulate the price
+    first_trade = 77026.57221815255
+    trade_state, trade_agent = oamm.execute_swap(
+        state=initial_state.copy(),
+        agent=initial_agent.copy(),
+        tkn_sell=trade_token,
+        tkn_buy=lp_token,
+        sell_quantity=first_trade
+    )
+
+    withdraw_state, withdraw_agent = oamm.execute_remove_liquidity(
+        state=trade_state.copy(),
+        agent=trade_agent.copy(),
+        quantity=trade_agent.holdings[('omnipool', lp_token)],
+        tkn_remove=lp_token
+    )
+
+    # second_trade = 250000.0
+    tkn_buy = trade_token
+    tkn_sell = lp_token
+    market_prices = {tkn: oamm.usd_price(initial_state, tkn) for tkn in initial_state.asset_list}
+    second_trade = (math.sqrt((
+        withdraw_state.lrna[tkn_sell] * withdraw_state.lrna[tkn_buy]
+        * withdraw_state.liquidity[tkn_sell] * withdraw_state.liquidity[tkn_buy]
+    ) / (market_prices[tkn_buy] / market_prices[tkn_sell])) - (
+        withdraw_state.lrna[tkn_buy] * withdraw_state.liquidity[tkn_sell]
+    )) / (withdraw_state.lrna[tkn_sell] + withdraw_state.lrna[tkn_buy])
+
+    final_state, final_agent = oamm.execute_swap(
+        state=withdraw_state.copy(),
+        agent=withdraw_agent.copy(),
+        tkn_sell=lp_token,
+        tkn_buy=trade_token,
+        sell_quantity=second_trade
+    )
+
+    profit = (
+            oamm.cash_out_omnipool(final_state, final_agent, market_prices)
+            - oamm.cash_out_omnipool(initial_state, initial_agent, market_prices)
+    )
+
+    print(f'profit analysis:')
+    print(f'agent LPs {lp_quantity} {lp_token} ({lp_percent * 100}%) '
+          f'at {initial_agent.share_prices["omnipool", lp_token]} USD spot price.')
+    print(f'agent sells {first_trade} {trade_token} for {lp_token}')
+    print(f'agent withdraws all {lp_token}')
+    print(f'agent sells {second_trade} {lp_token} for {trade_token}')
+    print(f'agent nets {profit}')
+
+    if profit > 0:
+        raise AssertionError(f'profit should be negative, but is {profit}')
+
+
+def test_double_add_manipulation_scenario():
```

Comment:
I think the other tests have been modified to do double add versions of the attack, right? Is it still worth having this test here?